src/Http/Middleware/Impersonate.php

```diff
@@ -2,16 +2,28 @@
 
 namespace A17\Twill\Http\Middleware;
 
-use Auth;
 use Closure;
+use Illuminate\Auth\AuthManager;
 
 class Impersonate
 {
+    /**
+     * @var AuthManager
+     */
+    protected $authManager;
+
+    /**
+     * @param AuthManager $authManager
+     */
+    public function __construct(AuthManager $authManager)
+    {
+        $this->authManager = $authManager;
+    }
 
     public function handle($request, Closure $next)
```

Comment:
This is not a part of this PR and issue, you can submit another one and point this out, so we don't handle all the issues in a single PR. It's also better for readability and control.

By the way, It should not. Laravel does not type hint this by default as it does not limit anyone from using custom Request classes. Please check this discussion, and especially this comment: 
https://laracasts.com/discuss/channels/general-discussion/laravel-5-middleware-handle-method-type-hinting-request?page=1#reply=67848

Reply:
I already read that one before creating this review. As [this comment](https://laracasts.com/discuss/channels/general-discussion/laravel-5-middleware-handle-method-type-hinting-request?page=1#reply=73515) points out, there is doc block about it but not a type hint. That's why I added "I think" because it is not clear. They could've added PSR-7 Request interface or nothing if all they want is allow other kinds of requests. If someone wants to use a custom request that doesn't implement/extend the PSR-7 or Laravel's Request class that he/she should rewrite all the middlewares because they will throw errors since the handle methods are written assuming the `$request` is `\Illuminate\Http\Request` because of `@param  \Illuminate\Http\Request  $request`. Anyways it is not related to this PR as you pointed.

Reply:
You are right about the PSR-7 requests. So, you can make a new issue :)